### PR TITLE
fix(work-items): use GitHub computed close-linkage for open-linked-PR filter (#654)

### LINKED ISSUE #642: babysit classifier: count_classified counts stale pipe-row classifications from non-thread review-summary comments
Found by the control tower's merge digest of PR #634 (shared babysit classifier, #534).

`count_classified` in `plugins/source-control/skills/babysit-prs/scripts/babysit_classify.py` counts self-authored classification pipe-rows in ANY comment, including PR-level review-summary comments that are never thread-resolved. A stale classification posted outside a review thread keeps counting after its finding is discounted by the thread-state guard, so `classified` can be inflated past a fresh open finding — a fail-open path for `READINESS_OK`.

Requires a specific posting pattern (classification rows in non-thread comments), so lower likelihood than the truncation defect fixed on #634, but same gate surface and same failure direction. Candidate fix: only count pipe-rows that live inside review-thread comments, or apply thread-state discounting to the comment container before counting.

Related: #534, PR #634 (digest thread), #465 (open-state discount).

### LINKED ISSUE #644: work-items(work): implementation subagents flip PRs to ready without confirming review findings were actually resolved
**Lane:** work · **Category:** operational (subagent behavior gap, observed twice in one session)

## Triggering examples

1. **PR #616** (issue #508, claude-ops `lanes` skill): after CI went green, the review bot flagged 3 MEDIUM findings. The subagent's first resume pass acknowledged and claimed to fix them, but the orchestrator's own diff/CI re-check found the same 3 findings still present — required a second explicit resume with a more forceful instruction ("verify by re-reading the diff yourself before pushing").
2. **PR #643** (issue #463, work-items open-linked-PR filter): after fixing an unrelated CI-blocking `typos` failure, the subagent flipped the PR to `ready-for-review` — but 2 REAL findings from an *earlier* review round (a regex missing a word-boundary lookbehind; an inaccurate "truncation-safe" claim in a doc) were still unaddressed in HEAD. The subagent's fix-and-flip pass apparently didn't re-check prior review rounds, only the CI failure it was explicitly asked to fix.

## Observed vs expected

- **Observed:** a subagent asked to "fix the CI failure, then flip to ready" (or similar scoped instruction) sometimes flips to ready without independently re-confirming that ALL outstanding review-bot findings — including ones from earlier rounds, not just the one it was explicitly pointed at — are actually resolved in the pushed HEAD.
- **Expected:** before flipping any PR from draft to ready, a subagent should re-fetch the full review-bot comment history (not just the most recent one) and self-verify every finding is addressed in the current diff, not just the specific issue it was asked to fix in that resume.

## Category

behavioral (subagent self-verification gap before a state-changing action)

## Suggested fix direction (not investigated in depth)

This may already be partially covered by #513 (role-separated mini-SDLC — implementer ≠ verifier) — a dedicated verifier role re-checking the full review history before flip would structurally prevent this class of gap, rather than relying on the implementer subagent's own diligence. Could also be a narrower fix: the orchestrator's dispatch/resume brief template could explicitly instruct "before flipping to ready, re-fetch ALL review-bot comments (not just the latest), and confirm each finding is resolved in the current diff" as a standing step, independent of #513's larger architecture change.

## Related

#616 (PR), #643 (PR), #513 (mini-SDLC role separation — likely supersedes a narrow fix here)

### LINKED ISSUE #654: work-items open-linked-PR filter: closing-keyword regex matches fenced-code-block examples
## Context

Surfaced by a Codex P2 review thread on PR #643 (thread `PRRT_kwDOTCGFQM6SLf9b`, `plugins/work-items/tools/work-item-tracker/adapters/github/README.md`). Filed per the triage rule as a design question rather than fixed inline.

## Problem

The "Open linked PRs" filter (new in #643) runs its closing-keyword `jq test(...)` over the **raw** PR body:

```
"(?i)(?<!\w)(?:close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved):?[ \t]+#\($n)(?![0-9])"
```

GitHub treats **fenced code blocks as inert** for auto-close linkage — documented in this same repo at `plugins/source-control/skills/pull-request/reference/create.md` ("Fenced code blocks ARE inert, so example snippets are safe"). The raw-body regex does not share that awareness: an open PR whose body contains a fenced example such as

    ```text
    Closes #<N>
    ```

matches `true` and drops the still-ready issue `#<N>` from the pickable frontier, even though GitHub will not auto-close it on merge.

## Why this is a design question, not a small inline fix

Making the match fence-aware in a `jq`/shell one-liner is not small or bounded: it requires tracking fence state (opening/closing fences, variable backtick counts, `~~~` fences, indented fences) before the keyword test. That is a real complexity-vs-precision tradeoff on a comparatively rare edge case (an open PR body carrying a fenced `Closes #<specific-N>` for the exact issue under test), so it does not qualify as a VALID-small-own-path inline fix under the triage rule.

## Root-cause candidate to evaluate (do not assume)

Rather than layering fence-stripping onto a raw-body heuristic, the authoritative signal may be GitHub's own computed linkage: the GraphQL `PullRequest.closingIssuesReferences` connection returns the issues GitHub will auto-close from the body. **To verify:** confirm whether `closingIssuesReferences` honors fence-inertness (and HTML-comment behavior) as expected; if so, it sidesteps *both* the fenced-block and the word-boundary heuristics entirely. This would be a design change to the filter, out of scope for the #643 triage.

## Related

- Note the same fence-blindness pattern also exists in the pre-create gate's line-anchored `KEYWORD_REGEX` (`create.md`); worth considering whether any chosen approach should be applied consistently across both matchers.

### LINKED ISSUE #667: process: issues get closed COMPLETED based on an assumption a linked PR merged, without verifying it actually did
Lane: babysit, observed 2026-07-20 on issue #653 (linked to PR #652, source-control plugin).

**Triggering example:** #653's own body states "After #504's fix landed (PR #652)..." and the issue was closed with `stateReason: COMPLETED` at 09:03:16Z. At that same moment, PR #652 was still OPEN and unmerged — it has since flipped to a real `CONFLICTING` mergeability state against `main`, meaning the fix #653 describes is not actually on `main` yet.

**Observed vs expected:** an issue that exists specifically to track a follow-up on an in-flight PR got closed as done before that PR merged, apparently on the assumption that "PR pushed" == "fix landed." Expected: closing an issue whose completion depends on a specific PR merging should gate on that PR's `state == MERGED`, not on the PR simply existing or having a commit pushed.

**Category:** process/operational. This is a new, distinct failure mode from the #645 race-condition family (concurrent-lane collisions producing contradictory *thread* state) — this one is about premature issue *closure* based on an unverified merge assumption, which could silently drop real follow-up work if nobody happens to notice the gap (as this lane did here only incidentally, while checking #652's merge readiness for an unrelated reason).

Not proposing a specific fix mechanism (issue-closing isn't a single script this lane could patch) — flagging the pattern so whichever lane/actor is closing issues based on PR activity can add a merged-state check before marking COMPLETED.

### PR DESCRIPTION
**Pull request by kyle-sexton**
## Summary

The `work-items` GitHub adapter's "Open linked PRs" mechanic (shipped in #463 / PR #643) decided
whether an issue already had an open closing PR by running a closing-keyword `jq` regex over the
**raw** PR body. GitHub treats fenced code blocks and HTML comments as inert for auto-close
linkage, but the raw-body regex did not: a PR body carrying a fenced `Closes #<N>` example matched
`true` and dropped the still-open, still-pickable issue `#<N>` from the `/work-items:work`
frontier. #643 shipped a partial `gsub` fence-stripper alongside the regex, but it recognized only
exactly-three backtick/tilde fences and silently missed four-or-more-backtick and indented fences —
exactly the fragile hand-roll #654 flagged as not-small.

## Fix

Replace the entire raw-body heuristic (the `gh pr list --search "<N> in:body"` prefilter, the
closing-keyword `jq` `test`, and the partial `gsub` fence-stripper) with GitHub's own computed
close-linkage, queried directly against the target issue number via the GraphQL
`Issue.closedByPullRequestsReferences` connection, keeping only `OPEN`-state PR nodes. This is the
root-cause fix #654 pointed toward: the signal is now GitHub's parsed issue linkage — the same set
GitHub renders in the issue sidebar and acts on at merge time — rather than a text match this
mechanic has to keep fence-, word-, and number-boundary-aware itself.

Consequences folded into the mechanic's rationale in the README:

- **Fenced code blocks and HTML comments are inert for free** — no fence-tracking heuristic to
  maintain, and the fence-blindness #654 filed is closed.
- **No word-boundary / number-boundary guards** — `#463` cannot collide with `#4630` / `#1463`
  because the reference is GitHub's parsed linkage, not a regex over raw text.
- **Base-branch correctness (behavior change, called out explicitly):** GitHub forms the close-link
  only for a PR that targets the default branch. A `Closes #<N>` on a non-default-base PR is
  ignored by GitHub and now no longer excludes its issue — where the retired raw-body regex counted
  it regardless of base. This is *more* correct per GitHub's real merge-time auto-close semantics.
- **Opt-out (`Refs #<num>`) stays intrinsic** — a reference-without-closing never enters the
  closing linkage, so it correctly does not exclude its issue.

Scope is held to the `work-items` adapter. The sibling fence-blind matcher noted in #654 — the
pre-create gate `KEYWORD_REGEX` in `plugins/source-control/skills/pull-request/reference/create.md`
— is **not** touched here: it is a different plugin, and it gates a PR body *before the PR exists*,
so it cannot use this issue-side GraphQL linkage. Whether it should adopt a different fence-aware
approach is left as a forward-reference follow-up, not folded into this adapter patch.

## Verification

Empirical research proving `Issue.closedByPullRequestsReferences` is queryable via `gh api graphql`
and returns GitHub's genuine auto-close linkage (open-state filtered) — not body text matches.

**Schema exists and is queryable** (live introspection, `gh api graphql`):

```
closedByPullRequestsReferences args:
  includeClosedPrs  Boolean = false  "Include closed PRs in results"
  first/last/after/before, orderByState, userLinkedOnly
```

**Positive cases — real prose closing keywords link, filtered to open PRs** (the exact mechanic
this PR ships, run live against this repo):

```
issue 642  -> true    (open issue; open PR #666 carries prose `Closes #642`)
issue 654  -> false   (this issue; no open closing PR)
issue 487  -> false   (closed issue; its closing PR #629 is MERGED, so filtered out by state=="OPEN")
issue 644  -> false   (baseline before this PR)
issue 667  -> false   (baseline before this PR)
```

**Negative isolation — fenced and HTML-comment closing keywords are excluded.** This PR body is
itself the live experiment. It carries a real `Closes #654` in prose (positive control: an open
issue closed via prose, in *this* draft PR, at *this* moment), plus two deliberately inert
fixtures — a closing keyword for open issue **#644 only inside a fenced code block**, and one for
open issue **#667 only inside an HTML comment**:

```text
inert fenced fixture (must NOT link #644):
Closes #644
```

Because #644 and #667 are open issues (baselines `false` above) and this same draft PR *does* link
#654 from prose, the only difference for #644/#667 is that their keyword sits in a fence / comment.
Live query results against this PR are recorded below.

Live results (this PR is #668, base branch `main`):

```
# PR-side — what GitHub links THIS PR as closing:
pullRequest(668).closingIssuesReferences  ->  [654]        # ONLY 654. #644 (fenced) and #667 (comment) excluded.

# Issue-side — the exact mechanic this PR ships:
issue(654).closedByPullRequestsReferences  ->  [{668, OPEN}]   # real prose `Closes #654` links (positive control)
issue(644).closedByPullRequestsReferences  ->  []              # fenced `Closes #644` inert — #668 absent
issue(667).closedByPullRequestsReferences  ->  []              # HTML-comment `Fixes #667` inert — #668 absent
```

Conclusion: within one PR body at one instant, GitHub linked the prose closing keyword (#654) and
excluded the identical keyword when it sat inside a fenced code block (#644) or an HTML comment
(#667). Since #644/#667 are open issues that this same draft PR would otherwise be capable of
linking, the fence / comment is the sole cause of exclusion. Both the fenced-example bug #654
reported and HTML-comment inertness are handled by `closedByPullRequestsReferences` with no
heuristic of our own.

Closes #654

## Related

- #654 — this issue (fence-blindness in the open-linked-PR closing-keyword regex).
- #463 / PR #643 — shipped the "Open linked PRs" mechanic (and its partial `gsub` fence-stripper)
  this PR replaces.

🤖 Generated with a Claude Code implementation subagent (issue #654)

### CODE REVIEW

**Comment by kyle-sexton** [comment]
Thanks — addressed in `361a3009`:

- **Finding 3 (missing `| tr -d '\r'`)** — fixed; the final pipeline now ends with `| tr -d '\r'` per the Gotchas rule.
- **Finding 1 (`first:20` silent cap)** — documented; added the page-bound note plus the cursor-pagination escape hatch (GraphQL analogue of the `--limit` note).
- **Finding 2 (`select(.state=="OPEN")` redundant with `includeClosedPrs:false`)** — respectfully, this is not redundant, and removing it would be a correctness regression. `includeClosedPrs:false` suppresses only `CLOSED` (unmerged) PRs; it still returns `MERGED` nodes. Verified live against issue #487 (its only closing PR, #629, is `MERGED`):

```
closedByPullRequestsReferences(first:20, includeClosedPrs:false).nodes  ->  [{"number":629,"state":"MERGED"}]
... | any                                    ->  true    # WITHOUT the state filter: false in-flight report
... | select(.state=="OPEN") | any           ->  false   # WITH the filter: correct
```

So the filter is load-bearing (an issue whose only closing PR merged to a non-default base, or that was reopened after a merge, must stay pickable). I kept it and documented the semantics inline instead of removing it.

**Comment by kyle-sexton** [comment]
## Post-green review classification

3 review rounds, all findings resolved by the worker itself before this pass:

| Finding | Severity | Classification | Action |
|---|---|---|---|
| Missing `\| tr -d '\r'` on GraphQL pipeline output | minor, Windows-actionable | VALID | Fixed |
| `first:20` pagination cap undocumented | minor | VALID | Fixed — documented with the "List items" `--limit` precedent |
| `select(.state=="OPEN")` called redundant | very minor | INVALID (review's own error) | Corrected — verified live against #487/#629 that `MERGED` nodes DO appear even with `includeClosedPrs:false`, making the filter genuinely load-bearing, not redundant |

Verdict from the final review pass: "ready to merge." CI green (16/16). Flipping to ready — worker completed all fix rounds but the final draft→ready step wasn't executed.

File: plugins/work-items/tools/work-item-tracker/adapters/github/README.md [hunk 1]
```diff
@@ -159,33 +159,58 @@ Match GitHub's issue-closing keyword set (`close`/`closes`/`closed`/`fix`/`fixes
...
 ```bash
-gh pr list --state open --search "<N> in:body" --json number,body --limit 1000 | tr -d '\r' \
-  | jq --arg n "<N>" 'any(.[]; (.body | gsub("(?s)(\\x60{3}|~{3}).*?\\1"; "")) | test(
-      "(?i)(?<!\\w)(?:close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved):?[ \t]+#\($n)(?![0-9])"))'
+OWNER_REPO=$(gh repo view --json owner,name -q '.owner.login + " " + .name' | tr -d '\r')
+gh api graphql \
+  -f query='query($owner:String!, $repo:String!, $n:Int!) {
+    repository(owner:$owner, name:$repo) {
+      issue(number:$n) {
+        closedByPullRequestsReferences(first:20, includeClosedPrs:false) { nodes { number state } }
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/work-items/tools/work-item-tracker/adapters/github/README.md]: Addressed in `cf5e0791`, taking the second option you offered — request the supported maximum with a documented fallback.

- Bumped `first:20` → `first:100` (GitHub GraphQL caps `first`/`last` at 100), cutting the reachable page-exhaustion window by 5×.
- Clarified the mechanism you flagged inline: because the connection retains `MERGED` nodes, the page bound counts *every* ever-linked closing PR, so a long merge/reopen history can consume slots ahead of the currently-open PR — not just a hypothetical 20+ *open* PRs.
- Kept the cursor-pagination escape hatch documented for the (not-realistic) >100 case.

I stopped short of implementing a full cursor-pagination loop in the copy-paste recipe: at the connection max, an issue would need 100+ distinct ever-linked closing PRs with the sole open one past position 100 — beyond any realistic frontier item. If you think the loop is warranted regardless, I'll defer that call to a human reviewer rather than expand the snippet unilaterally.

[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/work-items/tools/work-item-tracker/adapters/github/README.md]: **Triage disposition: VALID edge → filed, not fixed in-diff → #677.**

The finding is real: `closedByPullRequestsReferences(first:20, includeClosedPrs:false)` retains `MERGED` nodes, so an issue with >20 linked closing PRs whose lone `OPEN` PR sorts past the first page returns an all-`MERGED` page, the filter sees no `OPEN`, and reports the item as pickable — a false-keep that can double-dispatch work already in flight.

Kept it out of this PR's diff because the failure is **reversible**: double-dispatch produces a duplicate PR that is detectable and closeable, which places it outside the fix-in-diff carve-out (data-loss / fail-open-gate / security / silent-wrong-feeding-*irreversible*), and the trigger (>20 linked closing PRs, OPEN one paginated past page 1) is implausible in practice — as the README note already acknowledges. Root-cause remedy is cursor pagination (a `first:100` bump only moves the boundary; the connection has no server-side OPEN-state filter, so pagination is the only correct route). Captured with full repro + remedy in #677 (`priority: needs-triage`). Resolving.
